Shortcuts: Require modifier key to trigger shortcuts (`F`,`A`,`D`,`S`,`T`,`/`)

cdedreuille opened the pull request:
Closes #16022

## What I did

In this PR I'm modifying the default shortcuts to avoid having shortcuts being triggered unintentionally. I added `Alt` before the first 6 shortcuts transforming from `S` to `Alt + S`.

I also replaced `/` for search into `command + K` as we are exploring updating our search into a command palette and `command + k` is a well known shortcut for that.

## Checklist for Contributors

### Testing

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to communicate how to test your changes -->

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

<!-- Please include the steps to test your changes here. For example:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access X story

-->

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

cdedreuille committed:
Update shortcuts.ts

cdedreuille committed:
Modify some addons shortcuts

cdedreuille committed:
Update manager.spec.ts

cdedreuille commented on code/lib/manager-api/src/modules/shortcuts.ts:
```diff
@@ -125,12 +125,12 @@ type API_AddonShortcutLabels = Record<string, string>;
-  toggleNav: ['S'],
-  toolbar: ['T'],
-  search: ['/'],
+  fullScreen: [controlOrMetaKey(), 'F'],
+  togglePanel: [controlOrMetaKey(), 'A'],
+  panelPosition: [controlOrMetaKey(), 'D'],
+  toggleNav: [controlOrMetaKey(), 'S'],
+  toolbar: [controlOrMetaKey(), 'T'],
+  search: [controlOrMetaKey(), 'K'],
```
Thanks @ndelangen I updated them now.

cdedreuille committed:
Update preview-web.spec.ts

cdedreuille committed:
Update shortcuts.test.js

cdedreuille committed:
Update shortcuts.test.js

JReinhold reviewed (changes requested):
Code looks good, but it needs a migration note.

cdedreuille committed:
Update to alt

cdedreuille committed:
Update MIGRATION.md

cdedreuille committed:
Update Search.tsx

cdedreuille commented:
@JReinhold I updated the shortcuts to use `Alt` instead of `command for most of them except for search that will stay as `command + K` to match what others are doing with their command action. I also added migration guides as you suggested.

cdedreuille committed:
Update shortcuts

cdedreuille committed:
Merge branch 'next' into charles-update-shortcuts

cdedreuille commented:
It seems we have flaky E2E tests. Since all other tests passed, we discussed this with @JReinhold and he gave me approval to merge this PR directly since the E2E bug is very unlikely related to this issue.

adrian-cocoara-lgp commented:
Hello @cdedreuille,
These changes broke some of our Test Cases since the shortcuts keys have changed.
I checked the documentation if there is any way these shortcuts are configurable, but I can't find anything. 
Is there any way I can revert them to the values of Storybook 7? Thanks!
